pkg/block/azure/adapter.go

```diff
@@ -47,6 +60,15 @@ func NewAdapter(ctx context.Context, params params.Azure) (*Adapter, error) {
 	if preSignedExpiry == 0 {
 		preSignedExpiry = block.DefaultPreSignExpiryDuration
 	}
+	if params.Domain == "" {
+		params.Domain = BlobEndpointDefaultDomain
+	} else {
+		domain := strings.TrimSuffix(params.Domain, "/")
```

Comment:
Not blocking, but I think we should be more strict here.. Accepting things like `blob.core.usgovcloudapi.net/` is weird

Reply:
I can go either way, removing